baselines/fedavgm/README.md

```diff
@@ -0,0 +1,230 @@
+
+Regarding this architecture, the historical references mentioned on the FedAvg and FedAvgM papers are [this](https://web.archive.org/web/20190415103404/https://www.tensorflow.org/tutorials/images/deep_cnn) and [this](https://web.archive.org/web/20170807002954/https://github.com/tensorflow/models/blob/master/tutorials/image/cifar10/cifar10.py).
+
+Important to highlight the rationale with this CNN model stated on FedAvgM paper:
+
+> "*This model is not the state-of-the-art on the CIFAR-10 dataset, but is sufficient to show relative performance for the purposes of our investigation."*
+
+**The default CNN model in use on this baseline have a centralized accuracy of ~0.74. That is different from the reported 0.86 accuracy from the original FedAvg paper. But it is still sufficient to show the relative performance for the purposes of FedAvgM investigation.**
+
+**Dataset:** This baseline includes the CIFAR-10 and Fashion-MNIST datasets. By default it will run with the CIFAR-10. The data partition uses a configurable Latent Dirichlet Allocation (LDA) distribution (`concentration` parameter equals 0.1 as default) to create **non-iid distributions** between the clients. The understanding for this `concentration` (α) is that α→∞ all clients have identical distribution, and α→𝟢 each client hold samples from only one class.
+
+| Dataset | # classes | # partitions | partition method | partition settings|
+| :------ | :---: | :---: | :---: | :---: |
+| CIFAR-10 | 10 | `num_clients` | Latent Dirichlet Allocation (LDA) | `concentration` |
+| Fashion-MNIST | 10 | `num_clients` | Latent Dirichlet Allocation (LDA) | `concentration` |
+
+**Data distribution:** The following figure illustrates the use of multiple `concentration` values to generate the data distribution over 30 clients for CIFAR-10 (10 classes) - [source code](fedavgm/utils.py):
+
+![](_static/concentration_cifar10_v2.png)
+
+**Training Hyperparameters:**
+The following table shows the main hyperparameters for this baseline with their default value (i.e. the value used if you run `python main.py` directly)
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 10 |
+| number of rounds | 5 |
+| model | CNN |
+| strategy | Custom FedAvgM |
+| dataset | CIFAR-10 |
+| concentration | 0.1 |
+| fraction evaluate | 0 |
+| num cpus | 1 |
+| num gpus | 0 |
+| server momentum | 0.9 |
+| server learning rate | 1.0 |
+| server reporting fraction | 0.05 |
+| client local epochs | 1 |
+| client batch size | 64 |
+| client learning rate | 0.01 |
+
+### Custom FedAvgM
+In contrast with the original implementation from Flower v1.5.0, in this baseline we introduce the Nesterov accelerated gradient as part of the momentum applied in the server model. To use the original Flower implementation, use the argument `strategy=fedavgm`. By default, the custom implementation is used. But, you can also refer to it on the command line as `strategy=custom-fedavgm`.
```

Comment:
Maybe it's wroth stressing that having the Nesterov momentum actually is part of the original FedAvgM definition in the paper.

Reply:
Changed to the following:

> In contrast to the initial implementation found in Flower v1.5.0, our baseline incorporates the Nesterov accelerated gradient as a pivotal component of the momentum applied to the server model. It is worth emphasizing that the inclusion of Nesterov momentum aligns with the original definition of FedAvgM in the research paper.